client/state/form/actions.js

```diff
@@ -10,15 +16,67 @@ export const setFormProperty = ( field, value ) => {
 	};
 };
 
-export const nextStep = ( ) => {
-	return {
-		type: NEXT_STEP,
-	};
-};
-
 export const goToStep = ( stepIndex ) => {
 	return {
 		type: GO_TO_STEP,
 		step: stepIndex,
 	};
 };
+
+export const nextStep = () => ( dispatch, getState, { callbackURL, nonce, formSchema, formLayout } ) => {
```

Comment:
Maybe this action is _too_ complex, but that's because the actions responsibility (auto-advance the form) has a lot of rules (client-side validation, async server-side validation, etc). If you feel this is hard to read I could try separating this logic into smaller chunks.